Tests/SpackSmokeTest/CMakeLists.txt

```diff
@@ -6,7 +6,7 @@
 # against a currently installed version of AMReX. The resulting
 # executable can then be ran to test functionality.
 
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.22)
```

Comment:
Can you summarize again why we do this, just for documentation?

Should we generally raise the CMake dependency at some point? I wonder how confusing it is to have multiple components use different constrains.

Reply:
Bumping it up made the HIP stuff work for me on Spock. I recall searching and seeing something about HIP support improvements in 3.21, but let me double check and I will add it here. 

I wasn't very concerned about moving the version up here for exactly the reason you mentioned (smoke test). On top of that, Spack usually has easy access to recent versions of CMake. But I can/will test some lower versions to see when it breaks.

Reply:
Ok, in that case I would probably recommend to at least bump the version constrain for CMake in Spack itself during https://github.com/spack/spack/pull/28967 :)

Reply:
I was able to test it with CMake 3.20 and it worked when I turned off MPI. It turned out going from 3.21 to 3.22 fixed something about finding MPI compilers, but I suspect that is probably an issue specific to the system I'm testing on. What do you think about going to 3.20?

Reply:
Where did you run, on Crusher / a HPE/Cray?

I merged this to CMake 3.22, thus the Cray FindMPI nice-ness:
https://gitlab.kitware.com/cmake/cmake/-/merge_requests/6264

Reply:
You probably need 3.20 due to the other bugs we fixed that required `SHELL:` support with HIP.

So I would say: leave 3.14 for now on top.
- Check for CMake 3.20+ if either HIP or HIP+Fortran are requested.
- Check for CMake 3.17+ if CUDA is requested

Reply:
Yes, I was running on Spock. Looks like that fix was your doing :grin: , thanks for that! 

I added the tests and fatal_error messages as you suggested.